pkg/catalogv2/oci/client.go

```diff
@@ -0,0 +1,307 @@
+package oci
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"helm.sh/helm/v3/pkg/chart/loader"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/helm/pkg/provenance"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/sirupsen/logrus"
+	helmregistry "helm.sh/helm/v3/pkg/registry"
+	"helm.sh/helm/v3/pkg/repo"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/content/memory"
+	"oras.land/oras-go/v2/registry"
+	orasRegistryRemote "oras.land/oras-go/v2/registry/remote"
+	orasRegistryAuth "oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/errcode"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+// Client is a client which deals with helm charts
+// for fetching, storing that are inside OCI registries.
+type Client struct {
+	URL        string
+	repository string
+	registry   string
+	tag        string
+	insecure   bool
+	caBundle   []byte
+	plainHTTP  bool
```

Comment:
Effectively, `insecure` is the same as using HTTP, as the TLS guarantees are bypassed. To that point, we could ditch `plainHTTP` and use `insecure` (ideally renamed to `insecureSkipTLSVerification`) instead.

Reply:
@pjbgf In oras, there are two different things. I am using Oras library and so I am following their convention. Even Helm uses oras library

Insecure - https://github.com/oras-project/oras/blob/main/cmd/oras/internal/option/remote.go#L107C20-L107C25
plain HTTP - https://github.com/oras-project/oras/blob/main/cmd/oras/internal/option/remote.go#L109

Reply:
Already addressed in https://github.com/rancher/rancher/pull/43823/files#diff-b335c5d6572cc7826cac4d401e0534b1830b0744fdaaa5e22cee2dd03fd13bd7